### DISCUSSION TITLE
Error encountered while using Flutter

### DISCUSSION THREAD

**Discussion by Cyberblue1n** [community/community#141465] [Programming Help]
### Select Topic Area

Question

### Body

Hi there, I recently started learning app development using Flutter. However, when I try to run the **main.dart** file, an error occurs, it also give me a warning related to Java. I have tried to solve the issue, but nothing has worked so far. Can anyone tell me how to resolve this issue??

**warring msg given by java** --> ### Could not create task ':generateLockfiles'.
**Error msg:** Error: ### Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui show Canvas, Paint, Path, lerpDouble;

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Cyberblue1n ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by MikaStiebitz** [comment]
Hey, looking forward to helping you :)

**Could not create task ':generateLockfiles':**
Please check your java version with `java -version`
Flutter typically requires JDK 8 newer versions may work, depending on your Flutter setup.
If that does not help, you should checkout this [post ](https://stackoverflow.com/questions/73357365/build-gradle-error-could-not-create-tastk-generatelockfiles-in-flutter-proj)

**Dart library 'dart:ui' is not available on this platform:**
This error occurs because `dart:ui` is only available in Flutter's engine when running on a supported platform (like mobile, web, or desktop).

**I would also recommend that you execute this command:**
`flutter doctor`

**Comment by Cyberblue1n** [comment]
Hi, thank you for your response. 

**Could not create task ':generateLockfiles':**
As you told me to check the java version and also give me a stack over flow post. 
I go through that post and check my java version

### Java version:  22.0.2
But i don't have gradle install. so i install gradle distribution and also set the path as mentioned in the post. and i got the gradle
### Gradle version: 8.8
that should work regarding that post you have given. But still it failed 

the same error occured. 

**Dart library 'dart:ui' is not available on this platform:**

as you told to run flutter doctor. i run that, it found 1 issue regarding visual studio toolchain problem. But buddy, i don't want to create windows app now. Maybe in future, but for now i just want to create android app. 
But this kind error is given.

**Comment by Danidukiyu** [comment]
Hey Man,
Can you send a screenshot of the error message or error codes ?

**Comment by Cyberblue1n** [comment]
Hello Darkphoenix. Thank you for your response.
here i attached the error messages and screenshot of those errors. Please
check it out.

Thank you.
Regards,
Nilanjan Dey

> Message ID: <community/community/repo-discussions/141465/comments/10967260
> @github.com>
>

Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui show Canvas, Paint, Path, lerpDouble;
       ^
Context: The unavailable library 'dart:ui' is imported through these packages:

    package:demo_app => package:flutter => dart:ui
    ...

Detailed import paths for (some of) the these imports:

    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/binding.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/assertions.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/change_notifier.dart => package:flutter/src/foundation/memory_allocations.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/key.dart => package:flutter/src/foundation/diagnostics.dart => dart:ui
    package:demo_app/main.dart => package:flutter/material.dart => package:flutter/src/material/about.dart => package:flutter/foundation.dart => package:flutter/src/foundation/key.dart => package:flutter/src/foundation/diagnostics.dart => package:flutter/src/foundation/debug.dart => dart:ui       
    ...

/C:/flutter/packages/flutter/lib/src/material/app.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' as ui;
       ^
/C:/flutter/packages/flutter/lib/src/material/app_bar_theme.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
       ^
/C:/flutter/packages/flutter/lib/src/material/arc.dart:6:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
       ^
/C:/flutter/packages/flutter/lib/src/material/badge_theme.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
       ^
/C:/flutter/packages/flutter/lib/src/material/banner_theme.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
       ^
/C:/flutter/packages/flutter/lib/src/material/bottom_app_bar_theme.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
       ^
/C:/flutter/packages/flutter/lib/src/material/bottom_navigation_bar_theme.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
       ^
/C:/flutter/packages/flutter/lib/src/material/bottom_sheet_theme.dart:5:8: Error: Dart library 'dart:ui' is not available on this platform.
import 'dart:ui' show lerpDouble;
